### DISCUSSION TITLE
Programmatically change a representation type

### DISCUSSION THREAD

**Discussion by dhrblav** [molstar/molstar#301] [Q&A] [answered]
Hi,
How can I programmatically change a representation type to another, e.g. switch from Cartoon to Gaussian surface?
In the same vein: I can easily add new representations to components using plugin.managers.structure.component.addRepresentation(). How can I turn on and off each representation of a component?
Thank you!

**Comment by dsehnal** [comment, marked as answer]
Hi,

- you can use this approach to change any parameters about the visuals: https://github.com/molstar/molstar/issues/298
- ``setSubtreeVisibility`` can be used on ``plugin.state.data`` and ref of the component to toggle it's visibility.

Let me know if you need more help.

**Reply by dhrblav** [reply]
Thank you David. Will try these approaches.

**Reply by dhrblav** [reply]
Thanks again. The plugin.build().to().update() solution is quite powerful.

**Reply by dsehnal** [reply]
You can batch many updates in a single commit too etc. If you search the source for ``.build()`` there will be some examples of that.

**Comment by saurabhdea** [comment]
```
<body>
<div id="app-id" class="molstar-app"></div>
<script type="text/javascript">
molstar.Viewer.create("app-id", {"emdbProvider":"rcsb","layoutShowLeftPanel":true,"layoutShowRemoteState":false,"viewportShowAnimation":true,"pdbProvider":"rcsb","layoutShowSequence":false,"viewportShowSelectionMode":false,"layoutShowLog":false,"viewportShowExpand":true,"layoutShowControls":false,"layoutIsExpanded":false}).then(viewer => {
viewer.loadStructureFromUrl("./www/traj.xyz", "xyz")
});
</script>
</body>
```

in this body how can i use the gaussian surface as default generally structure render in cartoon but i want to open in gaussian surface. is it possible  ?
